src/db/reader_base.rs

```diff
@@ -0,0 +1,84 @@
+use std::cell::RefCell;
+use std::sync::mpsc;
+use std::thread;
+
+use neon::context::{Context, FunctionContext};
+use neon::event::Channel;
+use neon::handle::{Handle, Root};
+use neon::result::JsResult;
+use neon::types::{Finalize, JsBuffer, JsFunction, JsString, JsValue};
+
+use crate::db::types::{JsBoxRef, Kind, SnapshotMessage};
+
+pub struct ReaderBase {
+    tx: mpsc::Sender<SnapshotMessage>,
+}
+
+impl Finalize for ReaderBase {}
+impl ReaderBase {
+    pub fn js_new(mut ctx: FunctionContext) -> JsResult<JsBoxRef<Self>> {
+        // Channel for sending callbacks to execute on the sqlite connection thread
+        let (tx, rx) = mpsc::channel::<SnapshotMessage>();
+        let channel = ctx.channel();
+
+        let mut option = rocksdb::Options::default();
+        option.create_if_missing(true);
+        let path = ctx.argument::<JsString>(0)?.value(&mut ctx);
+        let db = rocksdb::DB::open_for_read_only(&option, path, false)
+            .or_else(|err| ctx.throw_error(&err))?;
```

Comment:
Instead of opening a new DB, can't we use existing DB connection?

Reply:
With the current implementation it is a little hard to use existing DB.

Reply:
With the current implementation it is a little hard to use existing DB.

I think we can make a new issue for that to work on it in the future.

Reply:
I think each request will create the new connection, maybe can you check how much impact it would make?

Reply:
Can you do something like this?
`let db = ctx.this().downcast_or_throw::<JsBoxRef<Database>, _>(&mut ctx)?;`

Reply:
Done.